sdk/eventhub/Microsoft.Azure.WebJobs.Extensions.EventHubs/tests/EventHubEndToEndTests.cs

```diff
@@ -240,16 +238,121 @@ public async Task EventHub_PartitionKey()
+            var producer = new EventHubProducerClient(EventHubsTestEnvironment.Instance.EventHubsConnectionString, _eventHubScope.EventHubName);
+            await producer.SendAsync(new EventData[] { new EventData(new BinaryData(_testId)) });
+
+            var (jobHost, host) = BuildHost<EventHubTestInitialOffsetFromStartEndJobs>(
+                builder =>
+                    builder.ConfigureServices(services =>
+                    {
+                        services.Configure<EventHubOptions>(options =>
+                        {
+                            options.InitialOffsetOptions.Type = "FromEnd";
+                            options.AddSender(_eventHubScope.EventHubName, EventHubsTestEnvironment.Instance.EventHubsConnectionString);
+                            options.AddReceiver(_eventHubScope.EventHubName, EventHubsTestEnvironment.Instance.EventHubsConnectionString);
+                        });
+                    }));
+            using (jobHost)
+            {
+                // We don't expect to get signalled as there should be no messages received with a FromEnd initial offset
+                bool result = _eventWait.WaitOne(Timeout);
+                Assert.False(result, "An event was received while none were expected.");
+
+                // send a new event which should be received
+                await producer.SendAsync(new EventData[] { new EventData(new BinaryData(_testId)) });
+                result = _eventWait.WaitOne(Timeout);
+                Assert.True(result);
+            }
+        }
+
+        [Test]
+        public async Task EventHub_InitialOffsetFromEnqueuedTime()
+        {
+            // Mark the time now and send a message which should be the only one that is picked up when we run the actual test host
+
+            var producer = new EventHubProducerClient(EventHubsTestEnvironment.Instance.EventHubsConnectionString, _eventHubScope.EventHubName);
+            for (int i = 0; i < 3; i++)
+            {
+                // send one at a time so they will have slightly different enqueued times
+                await producer.SendAsync(new EventData[] { new EventData(new BinaryData(_testId)) });
+                await Task.Delay(1000);
+            }
+            var consumer = new EventHubConsumerClient(
+                EventHubConsumerClient.DefaultConsumerGroupName,
+                EventHubsTestEnvironment.Instance.EventHubsConnectionString,
+                _eventHubScope.EventHubName);
+
+            var events = consumer.ReadEventsAsync();
+            _initialOffsetEnqueuedTimeUTC = DateTime.UtcNow;
+            int ct = 0;
+            await foreach (PartitionEvent evt in events)
+            {
+                if (++ct == 1)
+                {
+                    // use the timestamp from the first event for our FromEnqueuedTime
+                    _initialOffsetEnqueuedTimeUTC = evt.Data.EnqueuedTime;
```

Comment:
Can't you read the value from the EventData objects being sent?

Reply:
The EnqueuedTime is set by the service, or maybe I'm not following..

Reply:
But don't we update the EventData object after it's sent?